### PR TITLE
Support Scala 2.11 compilation and publishing

### PR DESCRIPTION
**Pull request by aslesarenko**

### CODE REVIEW

File: build.sbt [hunk 1]
```diff
@@ -6,11 +6,20 @@ organization := "org.scorexfoundation"
 
 name := "sigma-state"
 
+lazy val scala212 = "2.12.10"
+lazy val scala211 = "2.11.12"
+crossScalaVersions := Seq(scala212, scala211)
+scalaVersion := scala212
+
+javacOptions ++=
```
Comment (greenhat) [inline build.sbt]: @aslesarenko You can set different java source and targets for different scala versions. Something along these lines should do the job:
```
javacOptions in(Compile, compile) ++= (if (scalaBinaryVersion.value == "2.11") Seq("-target", "1.7", "-source", "1.7") else Seq("-target", "8", "-source", "8" ))
scalacOptions in(Compile, compile) ++= (if (scalaBinaryVersion.value == "2.11") Seq("-release", "7") else Seq("-release", "8"))
```

File: sigmastate/src/test/scala/sigmastate/helpers/SigmaTestingCommons.scala [hunk 2]
```diff
@@ -199,11 +198,11 @@ trait SigmaTestingCommons extends PropSpec
     r.positionLimit shouldBe positionLimitBefore
 
     // using ergo's(scorex) reader/writer
-    val w = new VLQByteStringWriter()
-    serializer.serializeWithGenericWriter(v, w)
-    val byteStr = w.result()
-    byteStr.nonEmpty shouldBe true
-    serializer.parseWithGenericReader(new VLQByteStringReader(byteStr)) shouldEqual v
+//    val w = new VLQByteStringWriter()
```
Comment (kushti) [inline sigmastate/src/test/scala/sigmastate/helpers/SigmaTestingCommons.scala]: commented out code is not good. Whether remove it or replace with another one.

**Comment by kushti** [comment]
@aslesarenko approved but please address a minor comment

File: sigmastate/src/test/scala/sigmastate/helpers/SigmaTestingCommons.scala (hunk 2 above)
Comment (aslesarenko) [reply sigmastate/src/test/scala/sigmastate/helpers/SigmaTestingCommons.scala]: removed.

File: .travis.yml [hunk 3]
```diff
@@ -16,27 +19,22 @@ stages:
...
   if: tag =~ ^v
 - name: snapshot
-  if: NOT tag IS present
+  if: "(NOT tag IS present) AND (type = push)"
 jobs:
   include:
   - stage: test
-    script: sbt -jvm-opts .travis.jvmopts test
+    script: sbt -jvm-opts .travis.jvmopts +test
     # enable scoverage after SpamSpecification timeout issue is resolved
```
[the remark this answers is not in the document]
Comment (greenhat) [reply .travis.yml]: Yep. Seems it's slide out of our attention. https://github.com/ScorexFoundation/sigmastate-interpreter/pull/560 has been ready for quite a while. I'll update it

**Pull request merged**